service/src/main/java/bio/terra/catalog/service/dataset/DatasetAccessLevel.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.catalog.service.dataset;
+
+import bio.terra.catalog.iam.SamAction;
+import java.util.EnumSet;
+import java.util.Set;
+
+public enum DatasetAccessLevel {
+  OWNER("owner", EnumSet.allOf(SamAction.class)),
+  READER("reader", EnumSet.of(SamAction.READ_ANY_METADATA)),
+  DISCOVERER("discoverer", Set.of());
+
+  private final String name;
```

Comment:
Might be worth commenting that the UI depends on this field's contents and changing it will require changing the UI code.